opentelemetry-sdk/src/trace/config.rs

```diff
@@ -143,43 +144,51 @@ impl Default for Config {
+                            otel_traces_sampler_arg = format!("{:?}", sampler_arg)
+                        );
                         Box::new(Sampler::ParentBased(Box::new(Sampler::TraceIdRatioBased(
                             1.0,
                         ))))
                     }
                 }
                 "parentbased_jaeger_remote" => {
-                    handle_error(
-                        Error::Other(String::from(
-                        "Unimplemented parentbased_jaeger_remote sampler. Falling back to default: parentbased_always_on"
-                    )));
+                    otel_warn!(
+                        name: "TracerProvider.Config.Sampler.Unsupported",
+                        message = "OTEL_TRACES_SAMPLER is set to 'parentbased_jaeger_remote' which is not implemented in this SDK version. Using fallback sampler: ParentBased(AlwaysOn). Configure an alternative sampler using OTEL_TRACES_SAMPLER"
+                    );
                     Box::new(Sampler::ParentBased(Box::new(Sampler::AlwaysOn)))
                 }
                 "jaeger_remote" => {
-                    handle_error(
-                        Error::Other(String::from("Unimplemented jaeger_remote sampler. Falling back to default: parentbased_always_on")));
+                    otel_warn!(
+                        name: "TracerProvider.Config.Sampler.Unsupported",
+                        message = "OTEL_TRACES_SAMPLER is set to 'jaeger_remote' which is implemented in this SDK version. Using fallback sampler: ParentBased(AlwaysOn). Configure an alternative sampler using OTEL_TRACES_SAMPLER"
+                    );
                     Box::new(Sampler::ParentBased(Box::new(Sampler::AlwaysOn)))
                 }
                 "xray" => {
-                    handle_error(
-                        Error::Other(String::from("Unimplemented xray sampler. Falling back to default: parentbased_always_on")));
+                    otel_warn!(
+                        name: "TracerProvider.Config.Sampler.Unsupported",
+                        message = "OTEL_TRACES_SAMPLER is set to 'xray'. AWS X-Ray sampler is not implemented in this SDK version. Using fallback sampler: ParentBased(AlwaysOn). Configure an alternative sampler using OTEL_TRACES_SAMPLER"
+                    );
                     Box::new(Sampler::ParentBased(Box::new(Sampler::AlwaysOn)))
                 }
                 s => {
-                    handle_error(
-                        Error::Other(format!("Unrecognised OTEL_TRACES_SAMPLER value: {}. Falling back to default: parentbased_always_on",
-                        s
-                    )));
+                    otel_warn!(
+                        name: "TracerProvider.Config.Sampler.InvalidSamplerType",
```

Comment:
TracerProvider.Config.Sampler.InvalidSamplerType
TracerProvider.Config.Sampler.Unsupported
TracerProvider.Config.Sampler.InvalidArgument

Lets use consistent naming. The 1st one has Sampler word repeated, while others don't.
A shorter name like `TracerProvider.InvalidSamplerType` should be sufficient IMHO.
Alternatively, name can be `TracerProviderInvalidConfiguration`, and message can explain what went wrong, and additional attributes can be used to describe the component being configured incorrectly.

Reply:
Have named it to:
TracerProvider.Config.InvalidSamplerArgument
TracerProvider.Config.UnsupportedSampler
TracerProvider.Config.InvalidSamplerType

Wanted to keep 3 levels of naming as done in other places. Let me know if this still doesn't look consistent.

Reply:
From consistency, this is better. Though I am unsure if the name is easily followable by end users. User may not have ever touched/used `Config` at all, and when they see 'Config' in event name, it may not be obvious. Good to follow up later.